lib/TransactionRunner.js

```diff
@@ -366,21 +374,15 @@ class TransactionRunner {
     this.ensureTransactionErrors(transaction);
     if (reason) { transaction.errors.push({ severity: 'warning', message: reason }); }
 
-    if (!transaction.test) { transaction.test = this.createTest(transaction); }
+    if (!transaction.test) {
+      transaction.test = this.createTest(transaction);
```

Comment:
As I understood we can't.

`this.createTest` runs only when there is no `transaction.test`. I am not aware of the exact order of the transaction/test instantiation, but in case of the skipped test `transaction.test` _already exists_. 

If we leave out the condition and always call `this.createTest()` we may lose some info from the original `transaction.test`. Of course, we can work around that and spread the test within the `this.createTest()` to ensure it can digest an undefined test:

```js
createTest(transaction) {
  const { test: existingTest = {} } = transaction

  return {
    ...existingTest,
    inheritedProp: transaction.originProp,
  }
}
```

Does this make sense to you?

Reply:
I would also not touch this as a part of a production fix. To provide the changes described above we would need to remove a `if (!transaction.test)` condition, and I have no idea what we may break by doing so. 

Perhaps, instead we can take this point into a future transaction runner re-design, as I find it way too complicated. Fingers crossed we can make it more plain and easier to understand.